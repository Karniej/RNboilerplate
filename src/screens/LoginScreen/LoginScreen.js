import React from 'react'
import PropTypes from 'prop-types'
import { ContentWrapper, Header, MainButton } from '../../components'

const LoginScreen = ({ navigation }) => {
  return (
    <ContentWrapper>
      <Header
        title='This is Login Screen'
      />
      <MainButton
        title='Log In'
        routeName='DrawerStack'
        navigation={navigation}
      />
    </ContentWrapper>
  )
}
LoginScreen.propTypes = {
  navigation: PropTypes.shape({ navigate: PropTypes.func.isRequired }).isRequired
}

export default LoginScreen
