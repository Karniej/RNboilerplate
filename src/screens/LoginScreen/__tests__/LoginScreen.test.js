import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import LoginScreen from '../LoginScreen'

configure({ adapter: new Adapter() })

test('Renders correctly, expect onpress to be defined, but not called', () => {
  const defaultProps = {
    navigation: { navigate: jest.fn() }
  }
  const wrapper = shallow(<LoginScreen
    {...defaultProps}
  />)
  expect(wrapper).toMatchSnapshot()
})
