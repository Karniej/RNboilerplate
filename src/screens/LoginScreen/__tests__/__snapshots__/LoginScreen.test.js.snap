// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly, expect onpress to be defined, but not called 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginScreen
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header
          title="This is Login Screen"
        />,
        <MainButton
          navigation={
            Object {
              "navigate": [MockFunction],
            }
          }
          resetStack={false}
          routeName="DrawerStack"
          title="Log In"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "This is Login Screen",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "navigation": Object {
            "navigate": [MockFunction],
          },
          "resetStack": false,
          "routeName": "DrawerStack",
          "title": "Log In",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header
            title="This is Login Screen"
          />,
          <MainButton
            navigation={
              Object {
                "navigate": [MockFunction],
              }
            }
            resetStack={false}
            routeName="DrawerStack"
            title="Log In"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "This is Login Screen",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "navigation": Object {
              "navigate": [MockFunction],
            },
            "resetStack": false,
            "routeName": "DrawerStack",
            "title": "Log In",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
